app/com/yetu/oauth2provider/oauth2/services/ImplicitGrantFlowService.scala

```diff
@@ -14,29 +16,17 @@ class ImplicitGrantFlowService[U](personService: IPersonService) {
   def validateRequest(implicit request: AuthorizationRequest): Future[YetuUser] = {
     parseHeaders { email =>
 
-      val maybeUser = personService.findYetuUser(email)
-
-      val maybeSuccess = for {
-        user <- maybeUser
-      } yield Future.successful(user)
-
-      maybeSuccess.getOrElse{
-        logger.debug(s"could not find the user with the email address [${email}] ")
-        Future.failed(ImplicitFlowException("User or key does not exist"))
```

Comment:
I would not use `getOrElse` to throw an Exception in the `None` case, in any case use pattern matching, which will also give you the cast:

```
personService.findByEmailAndProvider(email, EmailPasswordProvider.EmailPassword).map {
  case Some(u: YetuUser) => u
  case _ => throw new ImplicitFlowException("user not found")
}
```

I think it makes the intent clearer